src/fakeredis_cluster.erl

```diff
@@ -146,23 +146,21 @@ handle_cast(_, State) ->
 handle_call(cluster_slots, _From, State) ->
     ?LOG("Handling a CLUSTER SLOTS request"),
     {reply, {ok, create_cluster_slots_resp(State)}, State};
-handle_call({start_instance, Port}, _From, #state{max_clients = MaxClients} = State) ->
-    ?LOG("start_instance: ~p", [Port]),
-    fakeredis_instance_sup:start_acceptors(Port, MaxClients),
+handle_call({restart_node, Port}, _From,
+            #state{max_clients = MaxClients, options = Options} = State) ->
+    fakeredis_instance_sup:start_link(Port, Options, MaxClients),
     {reply, ok, State};
-handle_call({kill_instance, Port}, _From, State) ->
-    ?LOG("kill_instance: ~p", [Port]),
-    Pids = gproc:lookup_pids({p, l, {local, Port}}),
-    ?LOG("Stop port=~p, pids=~p", [Port, Pids]),
-    [exit(Pid, kill) || Pid <- Pids],
+handle_call({stop_node, Port}, _From, State) ->
+    Pids = gproc:lookup_pids({n, l, {fakeredis_instance_sup, Port}}),
+    [exit(Pid, normal) || Pid <- Pids],
```

Comment:
There is a great difference between killing an instance and gracefully stopping it. In the latter case, the process has a chance to close the socket properly. In the former case, it hasn't.

I think there are good reasons for testing both of these scenarios, but perhaps kill is the more important one, since it simulates a crash rather than a planned shutdown. Thus, I suggest keeping the kill function alongside the stop function.

However, instead of killing each connection process, I suggest killing the whole supervisor who owns the listen socket, or possibly both the connections and the supervisor.

Reply:
The kill_instance actually was more of a stop before as well..
The problem is that the exit(kill) also closes the sockets properly currently, I guess we need to change our spawned thread that catches trap_exits to make a kill work as expected. Maybe check the reason or similar.

Reply:
ignore previous comment, the previoius kill did actually not trigger the terminate functions. I misread the logs..
So I guess there is a difference...
The previous kill was more of a kill_connections than kill_instances/node
I can try to add kill node incl. the server socket and connections,,but I had problems with a escalating KILL to fakeredis_cluster when doing it...will try more

Reply:
Kill can't be trapped. "An exception to the above is if the exit reason is kill, that is if exit(Pid,kill) has been called. This unconditionally terminates the process, regardless of if it is trapping exit signals." (https://erlang.org/doc/reference_manual/processes.html#receiving-exit-signals)

I think however that our spawned process which closes the listen socket can still close the socket though, because when the parent is killed, the exit signal 'killed' is sent to other linked processes. So even though 'kill' can't be trapped, 'killed' can. Therefore, also the children which are handling the connections will also get the signal 'killed' and handle this if they trap exit, so if we want to simulate a crash, I guess we should exit(kill) them first and then kill the supervisor.

Reply:
was thinking about this, but was then thinking that the supervisor might restart them quickly...but since its a simple_one_for_one thats actually not the case, which is nice

Reply:
restart is currently set as `temporary`, so it should be fine I guess: "A temporary child process is never restarted"